packages/core/src/component/ToddleComponent.ts

```diff
@@ -251,17 +294,31 @@ export class ToddleComponent {
     for (const [actionKey, action] of Object.entries(
       this.component.onLoad?.actions ?? {},
     )) {
-      yield* getFormulasInAction(action, ['onLoad', 'actions', actionKey])
+      yield* getFormulasInAction(action, globalFormulas, [
+        'onLoad',
+        'actions',
+        actionKey,
+      ])
     }
     for (const [actionKey, action] of Object.entries(
       this.component.onAttributeChange?.actions ?? {},
     )) {
-      yield* getFormulasInAction(action, [
+      yield* getFormulasInAction(action, globalFormulas, [
         'onAttributeChange',
         'actions',
         actionKey,
       ])
     }
+    // Visit all component formulas in case they reference other formulas
+    for (const [key, componentFormula] of Object.entries(
+      this.component.formulas ?? {},
+    )) {
+      yield* getFormulasInFormula(componentFormula.formula, globalFormulas, [
```

Comment:
FYI we also didn't do this before. Component formulas could potentially reference other global formulas